From 4741076cae4f4284e1fff9a03f35475b8455af54 Mon Sep 17 00:00:00 2001
From: Imre Deak <imre.deak@nokia.com>
Date: Wed, 1 Apr 2009 14:36:39 +0200
Subject: [PATCH] DSS2: Add function to display object to get the backlight level

This is needed by an upcoming patch that changes the backlight
initialization to use the backlight level set by the bootloader.

Also add a field for the maximum backlight level.

Signed-off-by: Imre Deak <imre.deak@nokia.com>
---
 arch/arm/plat-omap/include/mach/display.h |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/arch/arm/plat-omap/include/mach/display.h b/arch/arm/plat-omap/include/mach/display.h
index 6288353..6b702c7 100644
--- a/arch/arm/plat-omap/include/mach/display.h
+++ b/arch/arm/plat-omap/include/mach/display.h
@@ -211,6 +211,8 @@ struct omap_dss_display_config {
 	int panel_reset_gpio;
 	int ctrl_reset_gpio;
 
+	int max_backlight_level;
+
 	const char *name;		/* for debug */
 	const char *ctrl_name;
 	const char *panel_name;
@@ -225,6 +227,7 @@ struct omap_dss_display_config {
 	void (*ctrl_disable)(struct omap_display *display);
 	int (*set_backlight)(struct omap_display *display,
 			int level);
+	int (*get_backlight)(struct omap_display *display);
 };
 
 struct device;
-- 
1.5.6.5

